javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/JavaSymbolSolver.java

```diff
@@ -158,51 +159,51 @@ public void inject(CompilationUnit destination) {
             }
         }
         if (node instanceof MethodCallExpr) {
-            SymbolReference<ResolvedMethodDeclaration> result = JavaParserFacade.get(typeSolver).solve((MethodCallExpr) node);
-            if (result.isSolved()) {
-                if (resultClass.isInstance(result.getCorrespondingDeclaration())) {
-                    return resultClass.cast(result.getCorrespondingDeclaration());
-                }
+            Optional<ResolvedMethodDeclaration> result = JavaParserFacade.get(typeSolver)
```

Comment:
given we're touching these lines already, please rename `result` (here and below) to `declaration` (or `correspondingDeclaration` or `decl` or similar)